custom_components/ha_creality_ws/__init__.py

```
@@ -468,12 +488,165 @@ async def request_cfs_info(call: ServiceCall) -> None:
+        """Write filament metadata to one CFS slot.
+
+        This is the only *write* path into the CFS. The payload shape comes from
+        @buzato's work in PR #75, confirmed against real hardware there but not
+        documented by Creality, so the outgoing value and the printer's echo are
+        both logged (see ``_log_material_echo``).
+        """
+        targets = _coordinators_for_devices(hass, call.data.get("device_id"))
+        if not targets:
+            raise ServiceValidationError(
+                "No Creality printer matched the selected device."
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)